components/selectDevice/Device.tsx

```diff
@@ -0,0 +1,53 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+import { useState, useTransition } from "react";
+
+import transferPlayback from "@/lib/device/transferPlayback";
+
+const Device = ({
+  device,
+  currentDeviceId,
+  isPlaying,
+}: {
+  device: SpotifyApi.UserDevice;
+  currentDeviceId: string;
```

Comment:
One quick thing — delete the whole `getPlaybackState`, just use the `is_active` property of SpotifyApi.UserDevice